webapp/channels/src/components/advanced_text_editor/send_button/send_post_options/core_menu_options.tsx

```diff
@@ -45,14 +80,25 @@ function CoreMenuOptions({handleOnSelect}: Props) {
                     values={{'9amTime': timeComponent}}
                 />
             }
+            className='core-menu-options'
+            {...extraProps}
         />
     );
 
-    const nextMonday = moment().
-        tz(userTimezone).
-        day(8). // next monday; 1 = Monday, 8 = next Monday
-        set({hour: 9, minute: 0, second: 0, millisecond: 0}). // 9 AM
-        valueOf();
+    function getNextWeekday(dateTime: DateTime, targetWeekday: number) {
```

Comment:
You can move this to be outside the component as its static.

Reply:
done here https://github.com/mattermost/mattermost/pull/29120/files#diff-3d804cbb710ab6df92df20bc7c511ba94dc7cbe9a8060150aa1cf03c49297ee9R85